src/Amazon.Extensions.CognitoAuthentication/CognitoDevice.cs

```diff
@@ -156,6 +156,16 @@ public string GetDeviceName()
         /// Gets the device from the Cognito service using the device key and user's access 
         /// token using an asynchronous call
         /// </summary>
+        public async Task GetDeviceAsync()
+        {
+            await GetDeviceAsync(default);
+        }
+
+        /// <summary>
+        /// Gets the device from the Cognito service using the device key and user's access 
+        /// token using an asynchronous call
+        /// </summary>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation</param>
         public async Task GetDeviceAsync(CancellationToken cancellationToken = default)
```

Comment:
The `= default` should be removed as it is a potential source of confusion as to which one it calls even though this compiles and it calls the no arg one.

Reply:
Removed the optional parameters. Agree it was confusing, I think this is why it's calling the one with no/fewer args:

https://csharpindepth.com/articles/Overloading
> When faced with a choice between a method which requires the compiler to fill in optional parameter values and one which doesn't, if the methods are otherwise "tied" (i.e. normal argument conversion hasn't decided a winner), overload resolution will pick the one where the caller has specified all the arguments explicitly

I can't find an exact line in the spec to link, maybe https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#12643-better-function-member
> Optional parameters with no corresponding arguments are removed from the parameter list